lucet-wasi-sdk/src/lib.rs

```diff
@@ -186,6 +185,120 @@ impl Link {
     }
 }
 
+pub trait AsLink {
+    fn as_link(&mut self) -> &mut Link;
+}
+
+impl AsLink for Link {
+    fn as_link(&mut self) -> &mut Link {
+        self
+    }
+}
+
+pub trait LinkOpts {
+    fn ldflag<S: AsRef<str>>(&mut self, ldflag: S);
+    fn with_ldflag<S: AsRef<str>>(self, ldflag: S) -> Self;
+
+    fn export<S: AsRef<str>>(&mut self, export: S);
+    fn with_export<S: AsRef<str>>(self, export: S) -> Self;
+}
+
+impl<T: AsLink> LinkOpts for T {
+    fn ldflag<S: AsRef<str>>(&mut self, ldflag: S) {
+        self.as_link().ldflags.push(ldflag.as_ref().to_owned());
+    }
+
+    fn with_ldflag<S: AsRef<str>>(mut self, ldflag: S) -> Self {
+        self.ldflag(ldflag);
+        self
+    }
+
+    fn export<S: AsRef<str>>(&mut self, export: S) {
+        self.as_link()
+            .ldflags
+            .push(format!("--export={}", export.as_ref()));
+    }
+
+    fn with_export<S: AsRef<str>>(mut self, export: S) -> Self {
+        self.export(export);
+        self
+    }
+}
+
+impl<T: AsLink> CompileOpts for T {
+    fn cflag<S: AsRef<str>>(&mut self, cflag: S) {
+        self.as_link().cflags.push(cflag.as_ref().to_owned());
+    }
+
+    fn with_cflag<S: AsRef<str>>(mut self, cflag: S) -> Self {
+        self.cflag(cflag);
+        self
+    }
+
+    fn include<S: AsRef<str>>(&mut self, include: S) {
+        self.as_link()
+            .cflags
+            .push(format!("-I{}", include.as_ref()));
+    }
+
+    fn with_include<S: AsRef<str>>(mut self, include: S) -> Self {
+        self.include(include);
+        self
+    }
+}
+
+pub struct Lucetc {
+    link: Link,
+    lucetc: lucetc::Lucetc,
+    tmpdir: TempDir,
+    wasm_file: PathBuf,
+}
+
+impl Lucetc {
+    pub fn new<P: AsRef<Path>>(input: &[P]) -> Self {
+        let link = Link {
+            input: input.iter().map(|p| PathBuf::from(p.as_ref())).collect(),
+            cflags: Vec::new(),
+            ldflags: Vec::new(),
```

Comment:
this skips around the `--no-threads` option that was just added in `Link::new` - I think you could phrase this as `Link::new(input)` and get that plus less code here

Reply:
You found my copypasta...

Should we just have `--no-threads` as a default option?

Reply:
that's what the hope of doing it in `new` was :) I didn't see any places `Link` was constructed outside that

Reply:
Ah, my branch must not be up to date then. Time to rebase, because yes we do want that option everywhere